File: app/src/main/java/zebrostudio/wallr100/ui/toppicks/TopPicksPresenterImpl.java
package zebrostudio.wallr100.ui.toppicks;

import javax.inject.Inject;

import zebrostudio.wallr100.data.DataManager;
import zebrostudio.wallr100.utils.FragmentTags;

/**
 * Listens to user actions from the UI ({@link TopPicksFragment}), retrieves the data and updates
 * the UI as required.
 */
public class TopPicksPresenterImpl implements TopPicksFragmentContract.TopPicksPresenter {

    private DataManager mDataManager;
    private TopPicksFragmentContract.TopPicksView mTopPicksView;

    @Inject
    public TopPicksPresenterImpl(DataManager dataManager){
        mDataManager = dataManager;
    }

    @Override
    public void bindView(TopPicksFragmentContract.TopPicksView view) {
        mTopPicksView = view;
    }

    @Override
    public void unbindView() {
        mTopPicksView = null;
    }

    @Override
    public void updateCurrentFragmentTag() {
        mDataManager.setCurrentlyInflatedFragmentTag(FragmentTags.MINIMAL_FRAGMENT_TAG);
    }
}

File: app/src/main/java/zebrostudio/wallr100/ui/toppicks/ToppicksPresenterImpl.kt
package zebrostudio.wallr100.ui.toppicks

import zebrostudio.wallr100.data.DataRepository
import javax.inject.Inject

class ToppicksPresenterImpl
@Inject constructor(private var dataRepository: DataRepository) : ToppicksContract.ToppicksPresenter {

  private var toppicksView: ToppicksContract.ToppicksView? = null

  override fun attachView(view: ToppicksContract.ToppicksView) {
    toppicksView = view
  }

  override fun detachView() {
    toppicksView = null
  }

  override fun updateFragmentName(name: String) {
    dataRepository.updateCurrentFragmentName(name)
  }

}